admin/app.py

```diff
@@ -152,6 +160,125 @@ def logout():
+        book.title = form.title.data
+        book.author = form.author.data
+        book.year = form.year.data
+        book.publisher = form.publisher.data
+        book.language = form.language.data
+        book.description = form.description.data
+        book.pages = form.pages.data
+        book.genres = form.genre.data
+        book.status = 'active'
+        try:
+            book.save()
+            return redirect('/book-storage')
+        except Exception as e:
+            traceback.print_exc(e)
+            return redirect('/add-book')
+    return render_template('add-book.html', form=form)
+
+@app.route('/import-file')
+def import_file():
+    return render_template('csv-import.html')
+
+
+@app.route('/import-file', methods=['POST', 'GET'])
+def uploadFiles():
+    try:
+        uploaded_file = request.files['file']
+        if uploaded_file.filename:
+            file_path = os.path.join(app.config['UPLOAD_FOLDER'], uploaded_file.filename)
+            uploaded_file.save(file_path)
+            with open(UPLOAD_FOLDER + '/' + uploaded_file.filename) as f:
+                file_data = json.load(f)
+            for example in file_data:
+                book = Book.from_json(json.dumps(example))
+                book.save(force_insert=True)
+        return redirect(url_for('import_file'))
+    except Exception as e:
+        print("Something went wrong!")
+        traceback.print_exc()
+        return redirect(url_for('import_file'))
+
+
+@app.route('/book-storage')
+def book_storage():
+    books = Book.objects()
+    return render_template('book-storage.html', books=books)
+
+@app.route('/book-active')
+def book_active():
+    books = Book.objects(status='active')
+    return render_template('book-active.html', books=books)
+
+
+@app.route('/book-inactive')
+def book_inactive():
+    books = Book.objects(status='inactive')
+    return render_template('book-inactive.html', books=books)
+
+
+@app.route('/book-storage/<string:_id>')
+def book_details(_id):
+    book = Book.objects.get(id=ObjectId(_id))
+    return render_template('book-details.html', book=book)
+
+
+@app.route('/book-update/<string:_id>', methods=['POST', 'GET'])
+def book_update(_id):
+    try:
+        book = Book.objects.get(id=ObjectId(_id))
+        form = UpdateBookForm(request.form)
+        if request.method == 'POST':
+            title = form.title.data
+            author = form.author.data
+            year = form.year.data
+            publisher = form.publisher.data
+            language = form.language.data
+            description = form.description.data
+            pages = form.pages.data
+            genres = form.genre.data
+            status = form.status.data
+            book.update(title=title, author=author, year=year, publisher=publisher, language=language,
+                        description=description, pages=pages, genres=genres, status=status)
+            return redirect('/book-storage')
+    except Exception as e:
```

Comment:
we need some useful message on the UI with the reason of failure, I assume that printing traceback in the server log is insufficient